src/client/clientlauncher.cpp

```diff
@@ -59,7 +59,8 @@ ClientLauncher::~ClientLauncher()
 
 	delete g_fontengine;
 
-	device->drop();
+	if (device)
```

Comment:
Not directly.

It seems that you changed the source code a bit too much for one try. You will get further chances to select related adjustments together with a desired patch granularity.